File: Graphs/floydWarshallAlgorithm.py
class Graph:

	INF = 9999
	V = 4

	graph = [[0, 3, INF, 2],
	         [4, 0, 6, 7],
	         [INF, 5, 0, INF],
	         [INF, INF, 1, 0]]

	def flyodWarshall(self):
		dist = self.graph
		v = self.V

		for k in range(0, v):
			for i in range(0, v):
				for j in range(0, v):
					dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])

		for i in range(0, v):
			for j in range(0, v):
				if dist[i][j] == self.INF:
					print("INF ", end = "")

				else:
					print(dist[i][j], " ", end = "")

			print()

g = Graph()

g.flyodWarshall()

File: Graphs/floydWarshallAlgorithm.java
import java.util.*;

class Main{

	final static int INF = 9999, V = 4;

	private static void floydWarshall(int[][] graph){
		int[][] dist = Arrays.copyOf(graph, V);

		//setting each vertex as intermediate
		for(int k = 0; k < V; k++){

			//for each vertex as source
			for(int i = 0; i < V; i++){
				//each vertex as destination for picked source
				for(int j = 0; j < V; j++){
					//updating shortest path
					dist[i][j] = Math.min(dist[i][j], dist[i][k] + dist[k][j]);
				}
			}
		}

		//print
		for(int i = 0; i < V; i++){
			for(int j = 0; j < V; j++){
				if(dist[i][j] == INF)
					System.out.print("INF ");
				else
					System.out.print(dist[i][j] + " ");
			}
			System.out.println();
		}
	}

	public static void main(String[] args){
		int[][] graph = {{0, 3, INF, 2},
	                     {4, 0, 6, 7},
	                     {INF, 5, 0, INF},
	                     {INF, INF, 1, 0}};

	    floydWarshall(graph);
	}
}